pkg/volume/glusterfs/glusterfs.go

```diff
@@ -656,6 +659,12 @@ func parseClassParameters(params map[string]string, kubeClient clientset.Interfa
 			cfg.user = v
 		case "restuserkey":
 			cfg.userKey = v
+		case "gid":
+			reqGid, err = strconv.Atoi(v)
```

Comment:
@liggitt Yes, as per comment (https://github.com/kubernetes/kubernetes/pull/35460#issuecomment-256951507) , once we set the annotation in PV, this would be automatically used by the POD , so that the POD will be running with that SGID and get access to the volume.